core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/DistinctAggregator.java

```diff
@@ -106,6 +108,17 @@ public void prepareRun(BatchAggregatorContext context) throws Exception {
     validate(context.getInputSchema(), conf.getFields(), context.getFailureCollector());
     context.getFailureCollector().getOrThrowException();
 
+    if (conf.getFields() == null) {
+      context.record(TransformLineageRecorderUtils.generateOneToOnes(
+        TransformLineageRecorderUtils.getFields(context.getInputSchema()),
+        "distinctAggregator",
+        "Removed duplicates in input records."));
+    } else {
+      context.record(TransformLineageRecorderUtils.generateOneToOnes(
+        Lists.newArrayList(conf.getFields()),
```

Comment:
use `new ArrayList<>(conf.getFields())`

Reply:
That won't work since conf.getFields() returns an Iterable, and there's no ArrayList constructor that converts an Iterable to a list.